workflow-gen/Program.cs

```diff
@@ -0,0 +1,123 @@
+                    options.RegisterActivity<NotifyActivity>();
+                    options.RegisterActivity<ReserveInventoryActivity>();
+                    options.RegisterActivity<ProcessPaymentActivity>();
+                    options.RegisterActivity<UpdateInventoryActivity>();
+                });
+            });
+
+            Task.Run(() => StartWorkflowGeneratorAsync(delayInMilliseconds));
+
+            using var host = builder.Build();
+            host.Run();
+
+
+        }
+
+        static internal async void StartWorkflowGeneratorAsync(int delayInMilliseconds)
+        {
+
+            TimeSpan delay = TimeSpan.FromMilliseconds(delayInMilliseconds);
+
+            DaprClientBuilder daprClientBuilder = new DaprClientBuilder();
+
+            DaprClient client = daprClientBuilder.Build();
+            var Counter = 0;
+            await client.SaveStateAsync<OrderPayload>("statestore", "Cars",  new OrderPayload(Name: "cars", TotalCost: 15000, Quantity: 100));
+
+            while (true)
+            {
+                Random random = new Random();
+                var num = random.Next(35);
+                OrderPayload orderInfo = new OrderPayload("Cars", num * 1000, num);
+                string orderId = Guid.NewGuid().ToString()[..8];
+
+                try
+                {
+                    Console.WriteLine("Executing workflow...");
+                    using (ExecutionCallTime.NewTimer())
+                    {
+
+                        await client.StartWorkflowAsync(
+                            workflowComponent: DaprWorkflowComponent,
+                            workflowName: nameof(OrderProcessingWorkflow),
+                            input: orderInfo,
+                            instanceId: orderId);
+
+                            GetWorkflowResponse state = await client.WaitForWorkflowStartAsync(
+                                instanceId: orderId,
+                                workflowComponent: DaprWorkflowComponent);
+
+                            Console.WriteLine("Your workflow has started. Here is the status of the workflow: {0}", state.RuntimeStatus);
+
+                            state = await client.WaitForWorkflowCompletionAsync(
+                                instanceId: orderId,
+                                workflowComponent: DaprWorkflowComponent);
+
+                            Console.WriteLine("Workflow Status: {0}", state.RuntimeStatus);
+
+                    }
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("Caught {0}", e.ToString());
+                    ExecutionFailureCount.Inc();
+                }
+
+                Counter++;
+                if (Counter > 4){
+                    await client.SaveStateAsync<OrderPayload>("statestore", "Cars",  new OrderPayload(Name: "Cars", TotalCost: 15000, Quantity: 100));
+                    Counter = 0;
+                }
```

Comment:
This bit is what allows the workflows to run forever, yeah?

Reply:
Yeah it will "restock" after 5 runs. Occasionally we run out of inventory and the workflow will run a bit differently.